provider/agent.go

```diff
@@ -259,31 +261,138 @@ func agentResource() *schema.Resource {
 				ForceNew:    true,
 				Optional:    true,
 			},
+			"resources_monitoring": {
+				Type:        schema.TypeSet,
+				Description: "The resources monitoring configuration for this agent.",
+				ForceNew:    true,
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"memory": {
+							Type:        schema.TypeSet,
+							Description: "The memory monitoring configuration for this agent.",
+							ForceNew:    true,
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
+										Type:        schema.TypeBool,
+										Description: "Enable memory monitoring for this agent.",
+										ForceNew:    true,
+										Required:    true,
+									},
+									"threshold": {
+										Type:         schema.TypeInt,
+										Description:  "The memory usage threshold in percentage at which to trigger an alert. Value should be between 0 and 100.",
+										ForceNew:     true,
+										Required:     true,
+										ValidateFunc: validation.IntBetween(0, 100),
+									},
+								},
+							},
+						},
+						"volume": {
+							Type:        schema.TypeSet,
+							Description: "The volumes monitoring configuration for this agent.",
+							ForceNew:    true,
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"path": {
+										Type:        schema.TypeString,
+										Description: "The path of the volume to monitor.",
+										ForceNew:    true,
+										Required:    true,
+										ValidateDiagFunc: func(i interface{}, s cty.Path) diag.Diagnostics {
+											if i.(string) == "" {
```

Comment:
This will probably be fine, but in general you should always check that a type-cast has succeeded, otherwise you can panic the provider.

Reply:
Better safe than sorry - you're completely right so I added a small check.